src/components/widgets/EditableList.vue

```diff
@@ -29,26 +29,39 @@ const entityNamePlural = computed(() => props.entitiesName || `${props.entityNam
 
 const isExpanded = ref(false);
 
-onMounted(() => (priorValues.value = deepCopyFromRaw(items.value)));
+onMounted(() => {
+  savedValues.value = deepCopyFromRaw(items.value);
+  priorValues.value = deepCopyFromRaw(items.value);
+});
 
 const indexBeingEdited = ref<number | null>(null);
 const isBeingEdited = (index: number) => indexBeingEdited.value === index;
 const priorValues = ref<any[]>([]);
-const modifiedItemIndexes = ref<number[]>([]);
+const savedValues = ref<any[]>([]);
+
+function modifiedItem(index: number) {
+  return items.value[index].uuid && !equals(items.value[index], savedValues.value[index]);
+}
 
 const itemsToSave = computed((): any[] => [
-  ...items.value.filter((item, index) => modifiedItemIndexes.value.includes(index)),
+  ...items.value.filter((item, index) => modifiedItem(index)),
   ...items.value.filter(({ uuid }) => !uuid),
 ]);
 
+watch(indexBeingEdited, (nextIndex, prevIndex) => {
+  if (prevIndex != null && nextIndex != null) {
+    commitEdit();
+    setEdit(nextIndex);
```

Comment:
Is there ever a situation where this would cause an infinite loop since `setEdit` modifies the value being watched?

Reply:
As far as I know, watch will not be triggered if the previous value is the same as the next value, and because Vue batches calls, its only called once.

1. We modify `indexBeingEdited`, this triggers the watcher with a value of 2 as an example
2. The watcher is run:
	1. `commitEdit` sets the value to `null`, the `watch` is set to run on next update
	2. `setEdit` sets the value to `2` again (same as the triggering value), the `watch` is was scheduled to run
3. Once finish, the batches updates are run, but because the value of indexBeingEdited is the same that triggered the initial update, it will not run the `watch` again.

I made this simple [playground](https://play.vuejs.org/#eNqNVE1v2zAM/SucMNQOaiQotlOXBmuHHLrDVnQDdvGhjkw7amXJ0EeawfB/HyUnaZIF2U6J+B75HinKHbtt2/HKI7tmU8uNaB1YdL6d5Uo0rTYOOjBYZfBaOL7c/Dxo6+ZVhdxBD5XRDSRUIslVrrhW1oFQJa7vUKh6XgqHJdyEIqnyUo4+BVrlFXdCqyAWKGnMGEGXK4BQQ0scS12nT0gopEQDp+F9F3n96GkUiMcy41UhPZJYBEioP9DiumnEIHdCKRngnVY0m5zXCZytTJzM0MceNYNU4drdh3AGrcFV/DuCm9mhhdpo36bJr1Blo1pBukuAd4MaXFzAruA2uOnmqJ9oBYQFpV2kJRl0p8z0US5kv82Hugqh7fXs0gagB5QW/6EaFGNlGs9Rn3NVpgT1ZChgk0klvV1eQ2J/K57QGR6x0SuaIrgl1QrOULmMTgiLMKIAEVAKWywkLVihymE3wRlR12gsuFfBMYMF8sKT2bfuInnTGZkYNnI6GdafFp8ODptWFg7pBDBdeOdofz5zKfjLTc6OL/nyMmez27KEq+lk4P5f3maDKPkRyc9h8vJq1nV/v6S+n04ICob3TLKMOUsjrkQ9frZa0XOOk80Ztd0KieZ7G96Azdn19tpyVkipX7/GmDMes22cL5G/nIg/23WI5ezBkF+zwpztMFeYGt0Az398o33ZAxtdeknsMyBNQEsfPA60O69Ksr3Hi27v40eJxvHTztcOld02FYzGZYv8nNEH6cuZ1t/sfhh/3Cxpz/o/a+G8Dg==) to understand it better, also a link to the [flush timing doc](https://vuejs.org/guide/essentials/watchers#callback-flush-timing)